src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,132 @@
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null); });
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user); });
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user); });
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user); });
+    }
+
+    @Test
+    void register_loginIsTooShort_notOk() {
+        user.setLogin(INVALID_CREDENTIAL_WITH_LENGTH_ZERO);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user); });
+        user.setLogin(INVALID_CREDENTIAL_WITH_LENGTH_THREE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user); });
+        user.setLogin(INVALID_CREDENTIAL_WITH_LENGTH_FIVE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user); });
+    }
+
+    @Test
+    void register_passwordTooShort_notOk() {
+        user.setPassword(INVALID_CREDENTIAL_WITH_LENGTH_ZERO);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user); });
+        user.setPassword(INVALID_CREDENTIAL_WITH_LENGTH_THREE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user); });
+        user.setPassword(INVALID_CREDENTIAL_WITH_LENGTH_FIVE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user); });
+    }
+
+    @Test
+    void register_invalidAge_notOk() {
+        user.setAge(INVALID_AGE_THREE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user); });
+        user.setAge(INVALID_AGE_ELEVEN);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user); });
+        user.setAge(INVALID_AGE_SEVENTEEN);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user); });
+    }
+
+    @Test
+    void register_theSameLoginExists_notOk() {
+        storageDao.add(user);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_validUserData_ok() {
+        User expectedUser = new User();
+        expectedUser.setLogin(VALID_LOGIN);
+        expectedUser.setPassword(VALID_PASSWORD);
+        expectedUser.setAge(VALID_AGE);
+        User actualUser = registrationService.register(expectedUser);
+        assertEquals(expectedUser, actualUser);

```

Comment:
let's also check that we can find the user by registered login
this test would pass if the register method would actually be

```
   @Override
    public User register(User user) { 
        return user;
    }
  ```
  So we need to test it does add to the DB